terraform/modules/application/ecs.tf

```diff
@@ -94,20 +94,75 @@ resource "aws_ecs_task_definition" "app" {
   }
 }
 
+resource "aws_ecs_task_definition" "sidekiq" {
+  #checkov:skip=CKV_AWS_336:using readonlyRootFilesystem to true breaks the app, as it needs to write to app/tmp/pids for example
+  family                   = "${var.prefix}-sidekiq"
+  cpu                      = var.ecs_sidekiq_task_cpu
+  execution_role_arn       = aws_iam_role.task_execution.arn
+  memory                   = var.ecs_sidekiq_task_memory #MiB
+  network_mode             = "awsvpc"
+  requires_compatibilities = ["FARGATE"]
+  task_role_arn            = aws_iam_role.task.arn
+
+  container_definitions = jsonencode([
+    {
+      name    = local.sidekiq_container_name
+      command = ["bundle", "exec", "sidekiq", "-t", "3"]
+      cpu     = var.ecs_sidekiq_task_cpu
+      environment = [
+        { Name = "EXPORT_PAAS_INSTANCE", Value = local.export_bucket_key },
+        { Name = "RAILS_ENV", Value = var.rails_env },
+        { Name = "REDIS_CONFIG", Value = "[{\"instance_name\":\"\",\"credentials\":{\"uri\":\"${var.redis_connection_string}\"}}]" },
+        { Name = "S3_CONFIG", Value = jsonencode(local.s3_config) }
+      ]
+      essential         = true
+      image             = var.ecr_repository_url
+      memoryReservation = var.ecs_sidekiq_task_memory * 0.75
+      user              = "nonroot"
+
+      logConfiguration = {
+        logDriver = "awslogs"
+        options = {
+          awslogs-group         = aws_cloudwatch_log_group.this.id
+          awslogs-region        = "eu-west-2"
+          awslogs-stream-prefix = var.prefix
+        }
+      }
+
+      secrets = [
+        { Name = "DATABASE_URL", valueFrom = var.database_connection_string_arn },

```

Comment:
Similarly to Rachael's comment above, if we don't need the S3_config (I can't see why we would, but may be wrong!) then I don't think we need this secret either?

Reply:
Ah yes so all the env vars I've listed here were everything I worked out sidekiq needed to run happily - including this! 😅(they're a subset of the ones we specify for the app task)